### PR TITLE
final version of bot

### PR DESCRIPTION
**Pull request by Afformativ**

### CODE REVIEW

File: bot/main.py [hunk 1]
```diff
@@ -0,0 +1,348 @@
...
+        "total_questions": len(questions),
+        "points": 0,
+        "wine_data": wine_data,
+        "help_used": False,
+    }
+
+    return ask_question(update, context, user_id)
+
+
+def ask_question(update: Update, context: CallbackContext, user_id: int):
```
Comment (Nattalli) [inline bot/main.py:126]:
```suggestion
def ask_question(update: Update, context: CallbackContext, user_id: int) -> None:
```

File: bot/main.py [hunk 2]
```diff
@@ -0,0 +1,348 @@
...
+    wine_name = context.user_data[user_id]["wine_data"][0]
+    question_text = question["question"]
+
+    message = f"Назва вина: {wine_name}\n\n{question_text}"
+
+    buttons = [
+        InlineKeyboardButton("Пропустити питання", callback_data="skip"),
+    ]
+
+    if context.user_data[user_id]["help_used"] is False:
```
Comment (Nattalli) [inline bot/main.py:143]:
```suggestion
    if not context.user_data[user_id]["help_used"]:
```

File: bot/main.py [hunk 3]
```diff
@@ -0,0 +1,348 @@
...
+        )
+    else:
+        update.message.reply_text(
+            message, reply_markup=InlineKeyboardMarkup.from_column(buttons)
+        )
+
+    return ANSWERING
+
+
+def answer_question(update: Update, context: CallbackContext):
```
Comment (Nattalli) [inline bot/main.py:160]:
```suggestion
def answer_question(update: Update, context: CallbackContext) -> None:
```

File: bot/main.py [hunk 4]
```diff
@@ -0,0 +1,348 @@
...
+            "партнерів за посиланням: http://127.0.0.1:8000/😉 А ще, даємо "
+            "Вам особисту знижку 10% на це вино. Ось Ваш промокод: PSIJD 🤫"
+        )
+
+    elif points == 4:
+        result_text = (
+            "А Ви майстер своєї справи😱 За такі приголомшливі результати "
+            "хочемо подарувати Вам знижку на це вино у розмірі 20% від "
+            "стандартної ціни. Використати знижку Ви можете за допомогою "
+            "промокоду: PKDOS, який діє у всіх магазинах-партнерах. Щоб "
```
Comment (Nattalli) [inline bot/main.py:249]: let's change promocode. For example, LWINE2023 would be better 😄

File: bot/main.py [hunk 5]
```diff
@@ -0,0 +1,348 @@
...
+            "дізнатись, в яких магазинах є це вино, перейдіть за посиланням: "
+            "http://127.0.0.1:8000/🍷🔥"
```
Comment (Nattalli) [inline bot/main.py:251]: can we make a link for our specific wine? We have a search by id in URL. Let's take a look what the link created our frontend developers. But it should be something like `'f"os.getenv("WEBSITE_LINK")'/wines/{wine.id}"`
(That is an example only)

**Review by Nattalli — CHANGES_REQUESTED** [review]

**Comment by kseniia-grishchenko** [comment]
what's the status? It would  be really nice to merge this already @Afformativ

**Pull request closed without merging**